angular_token_auth.js

```diff
@@ -297,13 +298,20 @@
             }
         };
 
-        //use cookies if available, otherwise try localstorage
-        if (storageMethods['cookie'].test() === true) {
+        if (MODULE_SETTINGS.STORAGE_METHOD === 'cookie') {
             return storageMethods['cookie'];
-        } else if (storageMethods['localStorage'].test()) {
+        } else if (MODULE_SETTINGS.STORAGE_METHOD === 'localstorage') {
```

Comment:
Could you change this to fetch `storageMethods[MODULE_SETTINGS.STORAGE_METHOD]` and see if anything is found? If not, you have your `else` on L305.

```js
var method = storageMethods[MODULE_SETTINGS.STORAGE_METHOD];
if (method) {
    return method;
} else {
    // Use the default 'auto' method...
    // ...
}
```